framework/application/display_window.cpp

```diff
@@ -0,0 +1,261 @@
+** the rights to use, copy, modify, merge, publish, distribute, sublicense,
+** and/or sell copies of the Software, and to permit persons to whom the
+** Software is furnished to do so, subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be included in
+** all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#include "application/display_window.h"
+#include "application/application.h"
+
+#include "util/logging.h"
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(application)
+
+DisplayWindow::DisplayWindow(DisplayContext* display_context) : display_context_(display_context)
+{
+    assert(display_context_ != nullptr);
+}
+
+VkResult DisplayWindow::SelectPhysicalDevice(const encode::InstanceTable* table,
+                                             VkInstance                   instance,
+                                             VkPhysicalDevice*            physical_device) const
+{
+    assert(table);
+    uint32_t physical_device_count = 0;
+    VkResult error                 = table->EnumeratePhysicalDevices(instance, &physical_device_count, nullptr);
+    if (error || !physical_device_count)
+        return error ? error : VK_ERROR_INITIALIZATION_FAILED;
+
+    std::vector<VkPhysicalDevice> physical_devices(physical_device_count);
+    error = table->EnumeratePhysicalDevices(instance, &physical_device_count, physical_devices.data());
+    if (error)
+        return error;
+
+    // choose a 1st discrete GPU or just the 1st one
+    *physical_device = physical_devices[0];
+    for (auto device : physical_devices)
+    {
+        VkPhysicalDeviceProperties properties;
+        table->GetPhysicalDeviceProperties(device, &properties);
+        if (properties.deviceType == VK_PHYSICAL_DEVICE_TYPE_DISCRETE_GPU)
+        {
+            *physical_device = device;
+            break;
+        }
+    }
+
+    return VK_SUCCESS;
+}
+
+VkResult DisplayWindow::SelectDisplay(const encode::InstanceTable* table,
+                                      VkPhysicalDevice             physical_device,
+                                      VkDisplayKHR*                display) const
+{
+    assert(table);
+    uint32_t num_displays;
+    VkResult error = table->GetPhysicalDeviceDisplayPropertiesKHR(physical_device, &num_displays, nullptr);
+    if (error || !num_displays)
+        return error ? error : VK_ERROR_INITIALIZATION_FAILED;
```

Comment:
Curly braces

Reply:
Hi Patrick,
Do you round every if() block?

Reply:
Yes

Reply:
done

Reply:
`CONTRIBUTING.md` states we try to follow the [Google Style Guide](https://google.github.io/styleguide/cppguide.html), which (unless I've misread it) prefers {} around every if and else block.